macosx/Controller.mm

```diff
@@ -341,18 +341,28 @@ + (void)initialize
     //cover our asses
     if ([NSUserDefaults.standardUserDefaults boolForKey:@"WarningLegal"])
     {
-        NSAlert* alert = [[NSAlert alloc] init];
-        [alert addButtonWithTitle:NSLocalizedString(@"I Accept", "Legal alert -> button")];
-        [alert addButtonWithTitle:NSLocalizedString(@"Quit", "Legal alert -> button")];
-        alert.messageText = NSLocalizedString(@"Welcome to Transmission", "Legal alert -> title");
-        alert.informativeText = NSLocalizedString(
-            @"Transmission is a file-sharing program."
-             " When you run a torrent, its data will be made available to others by means of upload."
-             " You and you alone are fully responsible for exercising proper judgement and abiding by your local laws.",
-            "Legal alert -> message");
-        alert.alertStyle = NSAlertStyleInformational;
-
-        if ([alert runModal] == NSAlertSecondButtonReturn)
+        auto bundleURL = (__bridge CFURLRef)[NSBundle.mainBundle bundleURL];
```

Comment:
A comment about why this conspicuous low-level API is used here would be great to have here in code and not only in GH PR discussion.

Also, we mostly avoid using C++'s `auto` in ObjC++ code of this project.